include/openenclave/advanced/allocator.h

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+/**
+ * @file allocator.h
+ *
+ * This file defines the interface that pluggable allocators must implement.
+ * See
+ * https://github.com/openenclave/openenclave/blob/master/docs/DesignDocs/PluggableAllocators.md
+ *
+ */
+
+#ifndef OE_ADVANCED_ALLOCATOR_H
+#define OE_ADVANCED_ALLOCATOR_H
+
+#include <openenclave/bits/types.h>
+
+/**
+ * @cond IGNORE
+ */
+OE_EXTERNC_BEGIN
+
+/**
+ * @endcond
+ */
+
+/**
+ * Callback for initializing the allocator.
+ *
+ * This function will be called by oecore at an appropriate time during enclave
+ * initialization. The allocator is expected to perform any global
+ * initialization in this function. This function is called prior to any global
+ * initialization.
+ *
+ * @param[in] heap_start_address The starting address of the enclave heap to be
+ * managed by the allocator.
+ * @param[in] heap_end_address The ending address of the enclave heap.
+ */
+void oe_allocator_init(void* heap_start_address, void* heap_end_address);
+
+/**
+ * Callback for cleaning up the allocator.
+ *
+ * This function will be called by oecore at an appropriate time during enclave
+ * termination. atexit functions will be executed prior to oe_allocator_cleanup.
+ * oe_allocator_cleanup may not register functions with atexit.
+ */
+void oe_allocator_cleanup(void);
+
+/**
+ * Callback for performing thread-specific initialization.
+ *
+ * This function will be called by oecore during initialization of an enclave
+ * thread, after the thread-local variables for the thread have been
+ * initialized.
+ *
+ * Note: Each ecall starts a new enclave thread that terminates when the ecall
+ * returns.
+ */
+void oe_allocator_thread_init(void);
+
+/**
+ * Callback for performing thread-specific cleanup.
+ *
+ * This function will be called by oecore just prior to the termination of
+ * enclave thread. Thread-specific exit functions (__cxa_thread_atexit
+ * functions) will be executed prior to oe_allocator_thread_cleanup.
+ *
+ * Note: Each ecall starts a new enclave thread that terminates when the ecall
+ * returns.
+ */
+void oe_allocator_thread_cleanup(void);
+
+/**
+ * Callback to allocate memory.
+ *
+ * This function will be called by oecore to implement malloc.
+ * oe_allocator_malloc must provide the same semantics as malloc.
+ * See: https://en.cppreference.com/w/c/memory/malloc
+ *
+ * @param[size] Number of bytes to allocate.
```

Comment:
```suggestion
 * @param[in] size Number of bytes to allocate.
```
incorrect doxygen syntax in the file.   Fix here and below.

Reply:
Good catch. Fixed.